### PR TITLE
Well.... quite a 'few' changes on the FE

### PR DESCRIPTION
**Pull request by hrishikeshpaul**
Like the title says lol.

There are a lot of lint errors, mostly the use of `any`. If you have the time you can try help me figure those out. But I've moved the files a little bit - needs a lot more work. Below are my rationale for the folders,

1. `_store`: I wanted to keep the store folder at the top because there will be quite a few interactions with it which is why the underscore. Contains all the actions/reducers.
2. `assets`: Keeping this in the root folder to be able to use absolute imports
3. `components`: This folder will have all the TSX files that are used more than once. Even the files that make use of the store are here. Maybe I should split them up - dumb components (those that are solely defined by their inputs and outputs) and smart components (that can use dispatch and useSelector)
4. `hooks`: For when we have custom hooks. I believe that the socket stuff can be hooks, although I'm not sure. I'm pretty new to hooks so still exploring.
5. `layout`: This is the godly layout that I created. Right now it has only one file with all the components/interfaces but I'm going to split it up. Its a 100% responsive and super customizeable.
6. `router`: Not sure what to put in this. Any suggestion? We can remove this if not required.
7. `services`: All the functions that make API calls.
8. `theme`: All the theme configurations. The only ones that are being used are `ChakraTheme.ts` and `_fonts.scss`. The others are for bootstrap. And although bootstrap is there, the current app relies 95% in Chakra. So all the variables (for colors, border radius etc) can be imported from the Chakra instance.
9. `util`: All utilities go here. 

#### How to test
1. Get the updates ofcourse
2. `npm ci`
3. `npm run start`
4. Go to port 3000

#### Important

I've changed the port of the frontend to 3000 cause 3001 was super weird. Make sure to update the `.env` file. Below I've posted the config,
```
SPOTIFY_REDIRECT_URI=http://localhost:3000/login
DEV_URL=http://localhost:3000
```

I've done quite a bit of research and played around it with myself to keep it super simple. Everyone suggests to avoid deep nesting of folders that I've tried to keep it as simple as possible. Let me know what you think!

### CODE REVIEW

**Comment by s-munro** [comment]
Thankfully there are only a couple of files changes.  Will take a look at this small PR tonight

**Comment by hrishikeshpaul** [comment]
@s-munro I sort of forgot to upload the main submodule.... (lol......)

**Comment by hrishikeshpaul** [comment]
There should be a new folder called `vb-fe`

**Comment by s-munro** [comment]
> There should be a new folder called `vb-fe`

To replace the `client` folder?

**Comment by hrishikeshpaul** [comment]
> To replace the client folder?

Yeah, so initially it was called `frontend` but I thought we should like productize the code. So all classnames, and root folders etc should be prefixed with `vb`. I'm assuming we'll rename our backend to
1. `vb-be`
2. `vb-api`
3. `vb-api-server`

Take your pick or suggest some haha

**Comment by s-munro** [comment]
> Yeah, so initially it was called `frontend` but I thought we should like productize the code. So all classnames, and root folders etc should be prefixed with `vb`. I'm assuming we'll rename our backend to
> 
> 1. `vb-be`
> 2. `vb-api`
> 3. `vb-api-server`
> 
> Take your pick or suggest some haha
Sounds good to me, I don't really mind.  If the FE folder is `vb-fe` then I suppose the BE should be `vb-be`

**Comment by hrishikeshpaul** [comment]
`vb-be` sounds funny lmao

**Comment by s-munro** [comment]
lol you're a goon.  What would you prefer it to be?  I really don't mind at all

**Comment by hrishikeshpaul** [comment]
Actually yeah keep it `vb-be` doesn't matter lmao

File: vb-fe/src/components/select/Select.tsx
Comment (s-munro) [inline vb-fe/src/components/select/Select.tsx:27]: I've been having to use refs quite a bit recently.  I've been handling it by initializing the ref with `null`, and typing it with the specific element type.  For example,
`const selectInputRef = useRef<HTMLSelectInput>(null)`

It looks like this ref is actually not used in this file though, so I think we can remove it.

File: vb-fe/src/components/select/SelectOption.tsx
Comment (s-munro) [inline vb-fe/src/components/select/SelectOption.tsx:17]: It looks like we could just add a tag type here for props.  I think we can spread the rest to the option element also, but am not entirely certain about this..
```
type Props = {
  label: string;
  score: number; // I'm not sure if this is a number or string in the FE, actually
}

export const SelectOption = ({label, score, ...rest}: Props) => {
<components.Option {...rest}>
  <Box>
    <span className="text-grey-1>#</span> <span>{label}</span>
    <Badge colorScheme="primary" mx={2}>
      {score}
    </Badge>
  </Box>
</components.Option>
}
```

File: vb-fe/src/components/select/SelectOption.tsx
Comment (s-munro) [inline vb-fe/src/components/select/SelectOption.tsx:43]: Is `data` in this case intended to be HTML data like className, etc?  That's how I've typically used a spread in these cases, but I'm not too sure what this `components.NoOptionsMessage` component takes in as props

File: vb-fe/src/components/select/SelectStyles.ts
Comment (s-munro) [inline vb-fe/src/components/select/SelectStyles.ts:7]: this is dope

File: vb-fe/src/components/sheet/Sheet.tsx
Comment (s-munro) [inline vb-fe/src/components/sheet/Sheet.tsx:9]: I don't think we need the `<any>` here, right?  It doesn't take any args.  Also, is the return for functional components inferred?  We might not need to type it

File: vb-fe/src/core/home/Home.tsx
Comment (s-munro) [inline vb-fe/src/core/home/Home.tsx:21]: Could just remove this

File: vb-fe/src/core/home/Home.tsx
Comment (s-munro) [inline vb-fe/src/core/home/Home.tsx:37]: Should add a `TODO` here so we don't forget about this

File: vb-fe/src/core/redirect/Redirect.tsx
Comment (s-munro) [inline vb-fe/src/core/redirect/Redirect.tsx:24]: We can remove this too before we merge

File: vb-fe/src/core/redirect/Redirect.tsx
Comment (s-munro) [inline vb-fe/src/core/redirect/Redirect.tsx:28]: Lol your linter replaced the apostrophe with double quotes, I'm dead

File: vb-fe/src/layout/Layout.tsx
Comment (s-munro) [inline vb-fe/src/layout/Layout.tsx:32]: Was this one needing to be outside of the other useEffect?  I see there's no dependency array so I'm curious

File: vb-fe/src/layout/Layout.tsx
Comment (s-munro) [inline vb-fe/src/layout/Layout.tsx:51]: This is pretty cool.  Something you might be interested in is the ability to do something like

```
export const Layout = () => { ... }

Layout.Footer = () => { ... }

Layout.Content = () => { ... }
```

It's definitely not necessary, but I'm not sure if you've ever seen this before, and I think it's pretty cool lol

File: vb-fe/src/util/Query.ts
Comment (s-munro) [inline vb-fe/src/util/Query.ts:12]: can remove this

**Review by s-munro — CHANGES_REQUESTED** [review]
IT'S BEEN DONE!   Lol this was great, great job.  I left a bunch of comments, apologies in advance

File: vb-fe/src/App.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/App.tsx:1]: there are no relative imports. yep thats the flow! check the `eslintrc.json` the rules are written there

File: vb-fe/src/_store/system/SystemTypes.ts
Comment (hrishikeshpaul) [reply vb-fe/src/_store/system/SystemTypes.ts:13]: i think we can just use interfaces because most places have interfaces, what to you think?

File: vb-fe/src/_store/user/UserActions.ts
Comment (hrishikeshpaul) [reply vb-fe/src/_store/user/UserActions.ts:47]: yeah you're right. probably show a tooltip

File: vb-fe/src/components/create/CreateRoom.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/components/create/CreateRoom.tsx:30]: yeah i know, but for now its just a placeholder till the backend is ready

File: vb-fe/src/components/create/CreateRoom.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/components/create/CreateRoom.tsx:83]: this should be an html change lement, let me look into it

Comment (hrishikeshpaul) [reply vb-fe/src/components/create/CreateRoom.tsx:83]: yeah lol i couldnt find it, might look into this in detail later

File: vb-fe/src/components/filters/Filters.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/components/filters/Filters.tsx:5]: oh yeah i need to change the working of the select. i'll remove this for nwo

File: vb-fe/src/components/loading/Loading.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/components/loading/Loading.tsx:15]: yeah i left it as a placeholder and forgot to update it.

the thing is when this onclose happens the loading in the state will be false because thats the way it hides

Comment (hrishikeshpaul) [reply vb-fe/src/components/loading/Loading.tsx:15]: but chakra requires an `onClose` method to be preset :(

File: vb-fe/src/components/navbar/Navbar.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/components/navbar/Navbar.tsx:30]: its there on the backend i just have to add it. i'll go into detail with these when i work on these components separately

File: vb-fe/src/components/select/Select.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/components/select/Select.tsx:27]: it is used right? 

Line 85: ref={selectInputRef}

Comment (hrishikeshpaul) [reply vb-fe/src/components/select/Select.tsx:27]: but year its not referenced anywhere i'll remove it

File: vb-fe/src/components/select/SelectOption.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/components/select/SelectOption.tsx:17]: so since this is react-select you cant assign custom props to it. you'll have to derive it from one of the props defined by react select - which I couldnt figure out

File: vb-fe/src/components/select/SelectOption.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/components/select/SelectOption.tsx:43]: yeah even i'm not too sure, but you have to derive it from react-select and the docs are so trashy i couldn't figure out

File: vb-fe/src/components/sheet/Sheet.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/components/sheet/Sheet.tsx:9]: oh it will take in some props later, so its just a placeholder rn.

and i think you need to add return types to elements, otherwise the linter complains lol

File: vb-fe/src/core/home/Home.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/core/home/Home.tsx:37]: yeah but like functions cannot e empty - eslint :(

File: vb-fe/src/layout/Layout.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/layout/Layout.tsx:32]: so this return takes place when the component is destroyed. its when the component is dismounted. i'll add comments for this file to make things clearer

File: vb-fe/src/layout/Layout.tsx
Comment (hrishikeshpaul) [reply vb-fe/src/layout/Layout.tsx:51]: yusss, i've seen this I can do this! in this PR or another one?

Comment (hrishikeshpaul) [reply vb-fe/src/layout/Layout.tsx:51]: i'll just do it in this one

File: vb-fe/src/_store/system/SystemReducer.ts
Comment (hrishikeshpaul) [reply vb-fe/src/_store/system/SystemReducer.ts:39]: yep can do

File: vb-fe/src/_store/system/SystemTypes.ts
Comment (s-munro) [reply vb-fe/src/_store/system/SystemTypes.ts:13]: Interfaces sound good for me, I don't have a preference

File: vb-fe/src/components/loading/Loading.tsx
Comment (s-munro) [reply vb-fe/src/components/loading/Loading.tsx:15]: Oh lol, we can just do like `() => null` or something.  I'm really not sure what else we could utilize the `onClose` for right now

Comment (hrishikeshpaul) [reply vb-fe/src/components/loading/Loading.tsx:15]: thats also not allowed LOL

**Review by s-munro — APPROVED** [review]
Looks great to me! 👍 Good to merge, in my opinion, when you're content with it

**Pull request merged**
